.github/workflows/unit-tests.yaml

```diff
@@ -60,16 +59,27 @@ jobs:
         name: install gcc
         run: apt update && apt install -y build-essential
 
-        
-      # run tests
-        
-      - name: run tests 
+      # build the project's environment
+      -
+        name: build environment
         env:
           test_dir: /github/workspace/${{ matrix.library }}
+          
+        # TODO: use case statement here, or add separate matrix.build_args
         run: |
-            if [[ "${{matrix.library}}" == "libs/architectures/" ]]; then
-                pinto -p $test_dir build -E wrapper
+            if [[ "${{ matrix.library }}" == "libs/architectures/" ]]; then
+                FLAGS="-E wrapper"
+            else
+                FLAGS=""
             fi
-            pinto -p $test_dir run pytest $test_dir/tests ${{matrix.args}}
+            echo $FLAGS

```

Comment:
Would you add a `printenv` somewhere here. So that next time it is easier to see the env vars.

Reply:
good idea done